db-engine/src/main/java/com/splicemachine/db/impl/sql/compile/OrNode.java

```diff
@@ -145,12 +157,14 @@ else if (vn instanceof BooleanConstantNode)
         
         if (bron.getLeftOperand() instanceof ColumnReference) {
 			cr = (ColumnReference) bron.getLeftOperand();
-			if (!bron.getRightOperand().isConstantOrParameterTreeNode())
-				return false;
+			if (!bron.getRightOperand().isConstantOrParameterTreeNode() ||
+				(bron.isInListProbeNode() && columnNumbers.size() > 1))
+			return false;
 		}
         else if (bron.getRightOperand() instanceof ColumnReference) {
 			cr = (ColumnReference) bron.getRightOperand();
-			if (!bron.getLeftOperand().isConstantOrParameterTreeNode())
+			if (!bron.getLeftOperand().isConstantOrParameterTreeNode() ||
+				(bron.isInListProbeNode() && columnNumbers.size() > 1))
```

Comment:
Mark, at the check, we may not have seen all the columns yet. Can the fix handle the following scenario:
a1 in (1,2,3) and b1 =1 or a1=4 and b1=2?

Reply:
Good find Yi.  I think I was contemplating a similar scenario last night, but it was quite late and I ended up missing it.  I pushed a commit to fix this case by making canConvertToInList process the first term twice.  Once with FirstTime == true, and then with FirstTime == false.  I also pushed some logic outside of the function into the function, because that's really where it belongs.

Reply:
Refined the fix to avoid calling canConvertToInList twice for the first OR term.